### ISSUE TITLE
SFTP Checksum mismatch due to case-sensitivity

### ISSUE THREAD

**Issue by pratheekrebala** [rclone/rclone#4518]
#### What is the problem you are having with rclone?

Argument `--checksum` does not work as expected with some SFTP backends because the server provides checksum in upper-case while rclone client expects the checksum to be lower-case.

This would cause the files to be transferred despite having an identical checksum.

#### What is your rclone version (output from `rclone version`)

```
rclone v1.52.2
- os/arch: linux/amd64
- go version: go1.14.4
```
#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux 64 bit.

####  Which cloud storage system are you using? (eg Google Drive)

SFTP server (through bitvise)

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

`rclone sync --checksum remote: local`

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

This has been edited to remove sensitive information.

```
2020/08/18 17:59:30 DEBUG : rclone: Version "v1.52.2" starting with parameters ["rclone" "-vvvv" "sync" "--dry-run" "--config" "[redacted config path]" "[redacted remote path]"]
2020/08/18 17:59:30 DEBUG : Using config file from "[redacted config path]"
2020/08/18 17:59:30 DEBUG : [redacted remote path]: New connection [removed ip]->[removed ip] to "SSH-2.0-8.43 FlowSsh: Bitvise SSH Server (WinSSHD) 8.43"
2020/08/18 17:59:31 DEBUG : fs cache: renaming cache item "[redacted remote path]" to be canonical "[redacted remote path]"
2020-08-18 18:28:39 DEBUG : sftp cmd = [redacted file name]
2020-08-18 18:28:39 DEBUG : sftp output = "2CAD20C19A8EB9BB11A9F76527AEC9BC [redacted file name]\r\n"
2020-08-18 18:28:39 DEBUG : sftp hash = "2CAD20C19A8EB9BB11A9F76527AEC9BC"
2020-08-18 18:28:39 DEBUG : [redacted file name]: MD5 = 2CAD20C19A8EB9BB11A9F76527AEC9BC ([redacted remote path])
2020-08-18 18:28:39 DEBUG : [redacted file name]: MD5 = 2cad20c19a8eb9bb11a9f76527aec9bc (Local file system at [redacted local path])
2020-08-18 18:28:39 DEBUG : [redacted file name]: MD5 differ
2020-08-18 18:28:39 NOTICE: [redacted file name]: Not moving as --dry-run
```

**Comment by pratheekrebala** [comment]
I believe this is identical to #4162

**Comment by pratheekrebala** [comment]
I also think that the DEBUG output on line #5 should provide full sftp command that is being run instead of just the file name...

**Comment by ncw** [comment]
This looks simple enough to fix - do you want to send a pull request to `strings.ToLower` the md5sum/sha1sum output?

**Comment by ncw** [comment]
Thanks @buengese :-)

This fix for this will be released in v1.53.2
